toolz/itertoolz.py

```diff
@@ -1055,3 +1056,14 @@ def random_sample(prob, seq, random_state=None):
 
         random_state = Random(random_state)
     return filter(lambda _: random_state.random() < prob, seq)
+
+
+def flat(level, seq):
+    """ Flatten a possible nested sequence by n levels """
+    if level < 0:
+        raise ValueError("level must be >= 0")
+    for item in seq:
+        if level == 0 or not hasattr(item, '__iter__'):
```

Comment:
Probably better to have outside the for loop:
```python
if level == 0:
    yield from seq
    return
```

Reply:
Thanks! That works really well.